opensfm/reconstruction.py

```diff
@@ -801,7 +630,10 @@ def resect_reconstruction(reconstruction1, reconstruction2, tracks_manager1,
 def add_observation_to_reconstruction(tracks_manager, reconstruction,
                                       shot_id, track_id):
     observation = tracks_manager.get_observation(shot_id, track_id)
+    n_bef = reconstruction.shots[shot_id].compute_num_valid_pts()
```

Comment:
Couldn't it hurt performance if we dot his `compute` thing at each observation we add ?

Reply:
It's not really necessary and can hurt performance with the linear datastructures. In the SfM case, we use a map and it's just a call to size().

I will remove it